hail/python/hail/table.py

```diff
@@ -1504,8 +1504,8 @@ def index(self, *exprs, product=False) -> 'Expression':
         ----------
         exprs : variable-length args of :class:`.Expression`
             Index expressions.
-        product : bool
-            If ``True``, value of expression is array of all matches.
+        all_matches : bool
```

Comment:
You renamed it to `all_matches`, but in the downstream code it's still `product`